src/script/media/BackgroundBlurrer/fragmentShader.glsl

```diff
@@ -0,0 +1,28 @@
+precision mediump float;
+
+// our texture
+uniform sampler2D u_image;
+// The segmenter bitmask
+uniform sampler2D u_mask;
+
+uniform vec2 u_textureSize;
+
+// the texCoords passed in from the vertex shader.
+varying vec2 v_texCoord;
+
+void main() {
+    vec2 onePixel = vec2(1.0, 1.0) / u_textureSize;
+    // FIXME currently we choose either the pixel of the original image or we blur it.
+    // This creates a rather rough transition. We need to find a way to smooth the edges and avoid pixelated transitions.
+    if (texture2D(u_mask, v_texCoord).r == 1.0) {
+        gl_FragColor = texture2D(u_image, v_texCoord);
+    } else {
+        vec4 colorSum;
+        for (int i = -8; i < 8; i++) {
+            for (int j = -8; j < 8; j++) {
+                colorSum += texture2D(u_image, v_texCoord + onePixel * vec2(i, j));
+            }
+        }
```

Comment:
This is the blur algo. We run this part of the shader only if the segmentation mask value is != 1